docs/docs/enclave-configuration.md

```diff
@@ -316,184 +316,72 @@ The path should be absolute or relative to the root of the enclave module.
-    ```
-
-### Example using the executable JAR
-
-1.  To avoid the need to set up a linux environment using a VM or container, download
-    [GraalVM](https://github.com/graalvm/graalvm-ce-builds/releases/tag/vm-21.2.0) for your operating system and
-    [install](https://www.graalvm.org/docs/getting-started/) it. Once you setup your GraalVM, ensure `native-image` is
-    installed:
+1. Create a `filter.json` file in the following path: `path/to/enclave/src/main/resources/META-INF/native-image/` 
+   with the following code to exclude unnecessary classes:
+   ```json
+   {
+   "rules": [
+   {"excludeClasses": "nonapi.**"},
+   {"excludeClasses": "com.r3.conclave.host.**"}
+   ]
+   } 
+   ```
+2. Download [GraalVM](https://github.com/graalvm/graalvm-ce-builds/releases/tag/vm-21.2.0) for your operating system 
+   and [install](https://www.graalvm.org/docs/getting-started/) it.
 
+3. Install the `native image`:
     ```bash
     $JAVA_HOME/bin/gu install native-image
     ```
 
-    Also enable [native-image-agent](https://www.graalvm.org/reference-manual/native-image/BuildConfiguration/#assisted-configuration-of-native-image-builds)
-    on the command line of the GraalVM `java` command:
-
-    ```bash
-    $JAVA_HOME/bin/java -agentlib:native-image-agent=config-output-dir=/path/to/enclave/src/main/resources/META-INF/native-image/ ...
-    ```
-
-1.  Add the [Shadow Gradle plugin](https://plugins.gradle.org/plugin/com.github.johnrengelman.shadow)
-    to the `plugins` section of the `host`'s `build.gradle`:
+4. Add the [Shadow Gradle plugin](https://plugins.gradle.org/plugin/com.github.johnrengelman.shadow) to the 
+   `plugins` section of the `host`'s `build.gradle` file:
 
     ```groovy
     plugins {
         id 'com.github.johnrengelman.shadow' version '6.1.0'
     }
     ```
 
-1.  Generate the shadow jar:
+5. Add the `EnclaveWebHost` main class to the `host`'s `build.gradle` file, *after* the plugins section:
 
+    ```groovy
+    project.mainClassName = "com.r3.conclave.host.web.EnclaveWebHost" 
+    ```
+   
+6. Generate the shadow jar:
     ```bash
     ./gradlew -PenclaveMode=mock host:shadowJar
     ```
+    This command creates an executable *shadow* JAR which contains all the host's and enclave's dependencies. You 
+    can find the shadow jar in the default location `host/build/libs/host-all.jar`.
 
-    Default location should be `host/build/libs/host-all.jar`.
-
-1.  Run the host with the agent enabled to generate the configuration files:
-
+7. Run the host with the agent enabled to generate the configuration files:
     ```bash
     $JAVA_HOME/bin/java -agentlib:native-image-agent=config-output-dir=/path/to/enclave/src/main/resources/META-INF/native-image/,caller-filter-file=/path/to/enclave/src/main/resources/META-INF/native-image/filter.json -jar /path/to/host/build/libs/host-all.jar
     ```
 
-1.  Trigger `enclave` logic by sending `client` requests.
-
+8. Trigger the `enclave` logic by sending a `client` request.
```

Comment:
I would clarify further "Trigger the execution of the enclave logic...`

Reply:
Done.